pkg/analyzer/analyzer.go

```diff
@@ -74,6 +76,19 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
+func getReportMessage(node ast.Node) string {
+	switch node.(type) {
+	case *ast.ForStmt:
+		return "nested context in loop"
+	case *ast.RangeStmt:
+		return "nested context in loop"
+	case *ast.FuncLit:
+		return "nested context in function literal"
+	default:
+		return "nested context"
```

Comment:
Can't you append the node type to the message?

Or shouldn't you return something like "unsupported nested context type"

Because by reading the code, I'm unsure this default value can happen

Reply:
thanks for the notice. updated the default return value. 

i also noticed you asked to add tests, i would like to let you know that i have already added them in `testdata/src/example.go`

Reply:
You are right. I wasn't sure, but it's not a _test.go file. If you confirm that tests are launched byusing this file, then I'm fine

Reply:
`pkg/analyzer/analyzer_test.go` tests the analyser using `golang.org/x/tools/go/analysis/analysistest` testing library. 

The comments in the form of `//want "<some lint message>"` are what used to test the expected output for that particular line.

Reply:
eg: https://github.com/Crocmagnon/fatcontext/pull/18#discussion_r1730171901